src/V3Task.cpp

```diff
@@ -581,13 +525,7 @@ class TaskVisitor final : public VNVisitor {
             refp->taskp()->fvarp()->user2p(outvscp);
         }
         // Replace variable refs
-        // Iteration requires a back, so put under temporary node
-        {
-            AstBegin* const tempp = new AstBegin(beginp->fileline(), "[EditWrapper]", beginp);
```

Comment:
How did you get rid of having the AstBegin wrapper?

Similar below.

Reply:
Only `AstVarRef` nodes are replaced, and none of the input nodes are that (i.e.: all `AstVarRef` are under the input list, so this works just fine. `backp()` is not required for `foreach`